Enviando emails e SMS:

Pode-se enviar emails e SMS automaticamente, com módulos apropriados.

SMTP:

SMTP é o protocolo utilizado para envio de emails, enquanto o recebimento é feito pelo IMAP.

Enviando emails:

Módulo utilizado: smtplib

In [ ]:
import smtplib

smtpObj = smtplib.SMTP('example@example.com', 587)
smtpObj.ehlo()
smtpObj.starttls()
smtpObj.login('myEmail@email.com','MinhaSenha')
smtpObj.sendmail('myEmail@gmail.com','Subject: Teste\n Teste')
smtpObj.quit()

Conectando-se a um servidor SMTP:

Gmail: smtp.gmail.com                                           
Outlook/ Hotmail: smtp-mail.outlook.com                         
Yahoo mail: smtp.mail.yahoo.com                                     
AT&T: smtp.mail.att.net (porta 465)                             
Comcast: smtp.comcast.net                                       
Verizon: smtp.verizon.net (porta 465)                           

Este endereço de conexão ao servidor será passado no argumento de smtplib.SMTP('smtp.gmail.com', 587)

In [1]:
import smtplib

smtpObj = smtplib.SMTP('smtp.gmail.com', 587)

Se o código acima não for executado, significa que o SMTP não suporta TLS na porta 587. Neste caso, usar: smtplib.SMTP_SSL() e porta 465.

Se não houver conexão com a internet,o retorno será: socket.gaierror

Em seguida, deve-se fazer conexão com o servidor SMTP, utilizando a função smtplib.ehlo()

In [2]:
import smtplib

smtpObj = smtplib.SMTP('smtp.gmail.com', 587)
smtpObj.ehlo()

(250,
 b'smtp.gmail.com at your service, [186.209.70.85]\nSIZE 35882577\n8BITMIME\nSTARTTLS\nENHANCEDSTATUSCODES\nPIPELINING\nSMTPUTF8')

Se a saída for o inteiro 250, então a conexão foi bem sucedida

Iniciando a criptografia TLS:

Passo obrigatório com a função smtpObj.starttls().

Se a conexão for feita por SSL, não é necessário este passo.

Fazendo login no servidor SMTP:

Usar senha e emails correspondentes para o login com a função smtpObj.login()

A senha deverá ser configurada e só poderá ser usada para o login com python.

In [5]:
import smtplib

smtpObj = smtplib.SMTP('smtp.gmail.com', 587)
smtpObj.ehlo()
smtpObj.starttls()
smtpObj.login()

(235, b'2.7.0 Accepted')

O retorno com o inteiro 235 signica que o login foi aceito

Enviando um email:

Para enviar email, utilizar a função smtpObj.sendmail() com três argumentos:

O email do remetente, email do destinatário e o corpo do email como string.

Esta string deve começar com 'Subject:\n'

Desconectando do servidor SMTP:

A função smtpObj.quit() é utilizado para a desconexão do servidor

In [10]:
import smtplib

smtpObj = smtplib.SMTP('smtp.gmail.com', 587)
smtpObj.ehlo()
smtpObj.starttls()
smtpObj.login()
subject = 'Teste'
body = 'Teste para saber ser o email vai ser enviado'
msg = f'Subject: {subject}\n\n{body}'
smtpObj.sendmail('luiz.tacin12@gmail.com','luiz.tacin12@usp.br',msg)
smtpObj.quit()

(221,
 b'2.0.0 closing connection x124-20020acae082000000b0038440f0f63dsm3567907oig.37 - gsmtp')

IMAP:

VER DEPOIS

Enviando mensagens de texto com o Twilio:

Twilio é um serviço de gateway SMS, mas não é o único.

Usar o módulo twilio

In [6]:
from twilio.rest import Client

accountSID = ''
authToken = ''
twilioCli = Client(accountSID, authToken)
myTwilioNumber = '+12765315740'
myCellPhone = ''

message = twilioCli.messages.create(
  body="Hello from Twilio",
  from_="",
  to=""
)

print(message.sid)

SMf98c912ce47a03e69210e9f2c17c7c67


Algumas informações podem ser tiradas:

message.status: ver o status da mensagem, como queued, etc                                                      
message.date_created: ver a data de envio               
message.sid: id da mensagem

Mensagem via whatsapp:

adicionar 'whatsapp:' em from e to

In [10]:
from twilio.rest import Client

account_sid = ''
auth_token = ''
client = Client(account_sid, auth_token)

message = client.messages.create(
  from_='whatsapp:',
  body='Your appointment is coming up on July 21 at 3PM',
  to='whatsapp:'
)

print(message.sid)

SM43ece7082fdbab10bfa6941ef4577a6e
